static/gsApp/components/profiling/alerts.tsx

```diff
@@ -378,6 +378,169 @@ export function ContinuousProfilingBetaSDKAlertBanner() {
+  subscription,
+}: ContinuousProfilingNoQuotaAlertBannerProps) {
+  const hasBudget = hasBudgetForContinuousOrUiProfilesHours(subscription);
+
+  // As long as they have a budget set, don't show this banner.
+  if (hasBudget) {
+    return null;
+  }
+
+  // For am2 plans, it's possible they're just using transaction profiles which uses performance units
+  // in this case, we don't want to ask them to set up on-demand budgets for no reason.
+  // So check to see if they're trying to send continuous/ui profiles before shwoing them this banner.
+  if (
+    !subscription.categories.profileDuration?.usageExceeded &&
+    !subscription.categories.profileDurationUI?.usageExceeded
+  ) {
+    return null;
+  }
+
+  const eventTypes: EventType[] = [
+    DataCategoryExact.PROFILE_DURATION,
+    DataCategoryExact.PROFILE_DURATION_UI,
+  ];
+
+  return (
+    <Alert.Container>
+      <Alert
+        system
+        type="warning"
+        showIcon
+        trailingItems={
+          <AddEventsCTA
+            organization={organization}
+            subscription={subscription}
+            buttonProps={{
+              priority: 'default',
+              size: 'xs',
+              style: {marginBlock: `-${space(0.25)}`},
+            }}
+            eventTypes={eventTypes}
+            notificationType="overage_critical"
+            referrer={`no-profiling-quota-${eventTypes.join('-')}`}
+            source="continuous-profiling-no-quota-banner"
+          />
+        }
+      >
+        {t(
+          'Continuous Profiling and UI Profiling requires you to set your On-demand budget.'
+        )}
+      </Alert>
+    </Alert.Container>
+  );
+}
+
+function ContinuousProfilingNoQuotaAlertAm3Banner({
+  organization,
+  subscription,
+}: ContinuousProfilingNoQuotaAlertBannerProps) {
+  const hasBudget = hasBudgetForContinuousOrUiProfilesHours(subscription);
+
+  // As long as they have a budget set, don't show this banner.
+  if (hasBudget) {
+    return null;
+  }
+
+  const eventTypes: EventType[] = [
+    DataCategoryExact.PROFILE_DURATION,
+    DataCategoryExact.PROFILE_DURATION_UI,
+  ];
+
+  return (
+    <Alert.Container>
+      <Alert
+        system
+        type="warning"
+        showIcon
+        trailingItems={
+          <AddEventsCTA
+            organization={organization}
+            subscription={subscription}
+            buttonProps={{
+              priority: 'default',
+              size: 'xs',
+              style: {marginBlock: `-${space(0.25)}`},
+            }}
+            eventTypes={eventTypes}
+            referrer={`no-profiling-quota-${eventTypes.join('-')}`}
+            source="continuous-profiling-no-quota-banner"
+          />
+        }
+      >
+        {t(
+          'Continuous Profiling and UI Profiling requires you to set your Pay-as-you-go budget.'
+        )}
+      </Alert>
+    </Alert.Container>
+  );
+}
+
+function hasBudgetForContinuousOrUiProfilesHours(subscription: Subscription): boolean {
```

Comment:
could be worth clarifying that this is just checking that the customer would be able to consume profile hours, not that the customer actually can (because it doesn't check that there's any budget remaining)